mkosi

```diff
@@ -2107,59 +2104,33 @@ def install_debian_or_ubuntu(args: CommandLineArguments,
         cmdline.append("--no-check-valid-until")
 
     cmdline += [args.release, root, mirror]
-
-    if not do_run_build_script and args.bootable and args.output_format == OutputFormat.gpt_btrfs:
-        cmdline[4] += ",btrfs-progs"
-
     run(cmdline, check=True)
 
-    # Debootstrap is not smart enough to deal correctly with alternative dependencies
-    # Installing libpam-systemd via debootstrap results in systemd-shim being installed
-    # Therefore, prefer to install via apt from inside the container
-    extra_packages = ['dbus', 'libpam-systemd', 'binutils']
-
-    # Also install extra packages via the secondary APT run, because it is smarter and
-    # can deal better with any conflicts
-    extra_packages.extend(args.packages)
+    # Install extra packages via the secondary APT run, because it is smarter and can deal better with any
+    # conflicts.
+    extra_packages = {"systemd", "systemd-sysv"}
+    extra_packages.update(args.packages)
 
     if do_run_build_script:
-        extra_packages.extend(args.build_packages)
-
-    # Work around debian bug #835628
-    os.makedirs(os.path.join(root, "etc/dracut.conf.d"), exist_ok=True)
-    with open(os.path.join(root, "etc/dracut.conf.d/99-generic.conf"), "w") as f:
-        f.write("hostonly=no")
+        extra_packages.update(args.build_packages)
 
     if not do_run_build_script and args.bootable:
-        extra_packages += ["dracut"]
+        if args.with_unified_kernel_images:
+            extra_packages.add("dracut")
+            extra_packages.add("binutils")
+        else:
+            extra_packages.add("initramfs-tools")
```

Comment:
I don't think we should change that. Installing dracut on all bootable systems has been aronud since at least 2018. Sure, Debian doesn't default to dracut (and Debian's dracut maintainer is a man with a hatred against systemd and tends to patch bashisms out of dracut), but initramfs-tools is a legacy project and starting to support that would be akin to supporting ifconfig over networkd. Being opinionated is ok. :)

Reply:
Ideally we'd still figure out kernel updates both with and without unified kernel images on debian though. Although I can't really be bothered more than this PR so I guess it'll have to wait until someone is interested in that.